### ISSUE TITLE
Небольшой импурв автозамены (хоткей + автообновление)

### ISSUE THREAD

**Issue by AntipodTX** [BladeMight/Mahou#287]
Можно ли добавить комбинацию клавиш, которая при включенной автозамене будет её отключать, а при отключенной - наоборот включать?
Также возможно ли добавить в настройки автообновление словаря автозамен, например, раз в неделю?
Спасибо!

**Comment by BladeMight** [comment]
> Также возможно ли добавить в настройки автообновление словаря автозамен, например, раз в неделю?

Я его не так часто обновляю, ипоследний раз было:
![2020 08 27 - 20'54'22_rect](https://user-images.githubusercontent.com/11931888/91477853-a7c35400-e8a7-11ea-994e-2f2edd981e3a.jpg) 22.12.18

>Можно ли добавить комбинацию клавиш, которая при включенной автозамене будет её отключать, а при отключенной - наоборот включать?

Можно, добавил в a483702, проверяй в latest-commit:

Для обновления до latest-commit:
1. Во вкладке "Обновления" поставить канал обновлений на LatestCommit(Beta), и обновиться как обычно(проверить, скачать обновление)nj
2. На странице: https://github.com/BladeMight/Mahou/releases/tag/latest-commit есть постоянно авто-обновляемые сборки. (если не знаешь какую брать - Release_x86_x64.zip)

❗ **Это скрытая функция, нужно вручную её включить/настроить**, в Mahou.ini найти:

```
[Hidden]
...
ToggleAutoSwitchHK=
...
```

И указать необходимую гор. клавишу здесь, формат такой: один или несколько модификаторов: `Ctrl` `Alt` `Shift` `Win`, и обязательный знак `|` и далее символ клавиши на английской раскладке: короче формат такой же как и функции сниппетов `__keyboard()` см: `Список названий клавиш(английская раскладка):` в [wiki/Сниппеты? Синтакс?](https://github.com/BladeMight/Mahou/wiki/%D0%A1%D0%BD%D0%B8%D0%BF%D0%BF%D0%B5%D1%82%D1%8B%3F-%D0%A1%D0%B8%D0%BD%D1%82%D0%B0%D0%BA%D1%81%3F). Пример:

```
[Hidden]
ToggleAutoSwitchHK=Ctrl+Alt|7
```
теперь сохранить Mahou.ini и нажать "применить" в Mahou, - это назначит гор. клавишу <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>7</kbd> которая будет переключать режим Автозамены. Однако помни что после перезапуска/нажатия применить состояние "Автозамены" сбрасывается на то которое было в ini/окне Mahou.

> И последнее - можно ли добавить настройку, что в программах-исключениях не работает только автозамена, а обычные клавиши конвертации и все остальные функции - работают, так как они удобные, но есть программы, где автоматика лишняя (консоль, редактор кода и.т.п.)?

Можно попробовать.

**Comment by BladeMight** [comment]
> Пример:

Еще пример, если например хочешь только на одну клавишу <kbd>F7</kbd> будут вот так:

```
[Hidden]
ToggleAutoSwitchHK=|F7
```

`|` обязательный он делит "модификаторы" от "клавиши".

**Comment by BladeMight** [comment]
@AntipodTX Ну так что, попробовал?

**Comment by AntipodTX** [comment]
@BladeMight да, спасибо, работает! только единственное - в приложении галка не меняется, т.е. открываешь приложение - автозамена выглядит включённой, а на самом деле нет, так как я отключил её комбинацией клавиш, но это некритично )
(извини что долго не отвечал - был в отпуске)
если вдруг надумаешь сделать третий пункт - готов тестировать )

**Comment by AntipodTX** [comment]
@BladeMight да, спасибо, работает! только единственное - в приложении галка не меняется, т.е. открываешь приложение - автозамена выглядит включённой, а на самом деле нет, так как я отключил её комбинацией клавиш, но это некритично )
(извини что долго не отвечал - был в отпуске)
если вдруг надумаешь сделать третий пункт - готов тестировать )
